kbi/0001/index.rst

```diff
@@ -0,0 +1,138 @@
+      status` (or `git diff`) and navigating to a directory which is
+      supposedly not a Git worktree, or opening such a directory in an
+      editor or IDE such as VS Code or Atom, will potentially run
+      commands defined by that other user.
+
+   Credit for finding this vulnerability goes to 俞晨东; The fix was
+   authored by Johannes Schindelin.
+
+A message for the commit which introduced the change to Git [1]_ is
+more verbose about the reasons - notably mentioning educational
+institutes as an example:
+
+::
+
+   commit 8959555cee7ec045958f9b6dd62e541affb7e7d9
+   Author: Johannes Schindelin <Johannes.Schindelin@gmx.de>
+   Date:   Wed Mar 2 12:23:04 2022 +0100
+
+       setup_git_directory(): add an owner check for the top-level directory
+       
+       It poses a security risk to search for a git directory outside of the
+       directories owned by the current user.
+       
+       For example, it is common e.g. in computer pools of educational
+       institutes to have a "scratch" space: a mounted disk with plenty of
+       space that is regularly swiped where any authenticated user can create
+       a directory to do their work. Merely navigating to such a space with a
+       Git-enabled `PS1` when there is a maliciously-crafted `/scratch/.git/`
+       can lead to a compromised account.
+       
+       The same holds true in multi-user setups running Windows, as `C:\` is
+       writable to every authenticated user by default.
+       
+       To plug this vulnerability, we stop Git from accepting top-level
+       directories owned by someone other than the current user. We avoid
+       looking at the ownership of each and every directories between the
+       current and the top-level one (if there are any between) to avoid
+       introducing a performance bottleneck.
+       
+       This new default behavior is obviously incompatible with the concept of
+       shared repositories, where we expect the top-level directory to be owned
+       by only one of its legitimate users. To re-enable that use case, we add
+       support for adding exceptions from the new default behavior via the
+       config setting `safe.directory`.
+       
+       The `safe.directory` config setting is only respected in the system and
+       global configs, not from repository configs or via the command-line, and
+       can have multiple values to allow for multiple shared repositories.
+       
+       We are particularly careful to provide a helpful message to any user
+       trying to use a shared repository.
+
+Later changes, citing feedback from users who have a very large list of
+shared repositories, introduced the possibility to set the value of the
+config option to ``*``, implying that all directories are safe.
+
+Recommendation
+--------------
+
+Taking into consideration the information above, DataLad users should
+use their own judgment in setting the ``safe.directory`` config option.
```

Comment:
As I realised in the discussion below, although the commit quoted from the Git project refers mostly to shared repos, it is only the clone location that matters for the (un)safe directory. So a bare repo (shared or non-shared) in any kind of shared ("scratch") space would trigger the warning. So the option must be set, or clone won't happen.

I rephrased the recommendation to emphasize that it is a declaration of trust.

I also added a single commit rephrasing your comment about shared repositories above - although in (new) context this may seem slightly tangential.